Refactor the Dockerfile

andyduong1920 opened the pull request:
## What happened

Just refactor the Dockerfile
 
## Insight
 
## Proof Of Work

The test is passed

hanam1ni commented:
I tried using `\\` and it worked
```elixir
test "copies the Dockerfile", %{
      project: project,
      test_project_path: test_project_path
    } do
      in_test_project(test_project_path, fn ->
        Addons.Docker.apply(project)

        assert_file("Dockerfile", fn file ->
          assert file =~ """
                        RUN cd assets && \\
                        \t\tnpm ci --progress=false --no-audit --loglevel=error && \\
                        """
        end)
      end)
    end
```

andyduong1920 commented:
> I tried using `\\` and it worked
> 
> ```elixir
> test "copies the Dockerfile", %{
>       project: project,
>       test_project_path: test_project_path
>     } do
>       in_test_project(test_project_path, fn ->
>         Addons.Docker.apply(project)
> 
>         assert_file("Dockerfile", fn file ->
>           assert file =~ """
>                         RUN cd assets && \\
>                         \t\tnpm ci --progress=false --no-audit --loglevel=error && \\
>                         """
>         end)
>       end)
>     end
> ```

Thanks @hanam1ni I have updated it in https://github.com/nimblehq/elixir-templates/pull/94/commits/3b61cffc713404135c95c5e2637f17c8a2c91660 and updated the PR's description ✅

byhbt commented on priv/templates/nimble_template/Dockerfile.eex:
```diff
@@ -1,52 +1,55 @@
-RUN npm run --prefix ./assets deploy
-RUN mix phx.digest
+RUN cd assets && \
+		npm ci --progress=false --no-audit --loglevel=error && \
```
hi @andyduong1920 !

I have a question here, I have a problem with this line when I try to build the Docker image, 
In `assets`, I used `yarn` to install then dependencies, then it will fail here right? Because there is no `npm-package.lock` if you use `yarn`.
So we have to use `npm` to install the packages, is it correct?

andyduong1920 replied:
Yup correct @byhbt, the Phoenix generator only support npm (package-lock.json) for now. ref https://hexdocs.pm/phoenix/up_and_running.html#content
